### DISCUSSION TITLE
Description of location-lookup API

### DISCUSSION THREAD

**Discussion by gurupras** [photoprism/photoprism#4580] [Q&A] [answered]
In the interest of being completely private (personal photos, after all), I blocked my docker container from being able to access the internet. Instead, I have another container that acts as a reverse-proxy for photoprism and provides access from LAN/internet.

That said, I would still love to have some geo..tagging?
It looks like API requests are made to `https://places.photoprism.app/v1/location/<hash>`. For example:
https://places.photoprism.app/v1/location/12a4bc47c90c
This returns
> {"id":"s2:12a4bc47c90c","name":"Vallençana Baixa","street":"Carretera de Badalona","postcode":"08110","category":"","timezone":"Europe/Madrid","lat":41.478317,"lng":2.195216,"place":{"id":"es:9HOSc35zDTMJ","label":"Montcada i Reixac, Catalunya, Spain","district":"","city":"Montcada i Reixac","state":"Catalunya","country":"es","keywords":""},"events":[],"licence":"Data © OpenStreetMap contributors, ODbL 1.0, see https://osm.org/copyright"}

What is this format and what would it take from my end to host this myself or make the API calls myself?

**Comment by lastzero** [comment, marked as answer]
The data structure (request and response fields) can be found in our publicly available source code here on GitHub. It is also described in our developer guide. Location data can be obtained for free from OpenSteetMap.

The effort for running a reverse geocoding service will depend on your runtime environment as well as performance and update requirements. Server hardware should have [at least 128 GB of RAM and a 4 TB SSD](https://nominatim.org/release-docs/latest/admin/Installation/#hardware).

License prices for self-hosted world maps can be found, for example, at [www.maptiler.com/data/pricing/](https://www.maptiler.com/data/pricing/).
